exe/floe

```diff
@@ -7,15 +7,32 @@ require "optimist"
 opts = Optimist.options do
   version "v#{Floe::VERSION}\n"
   opt :workflow, "Path to your workflow json", :type => :string, :required => true
-  opt :inputs, "JSON payload to input to the workflow", :type => :string, :default => '{}'
-  opt :credentials, "JSON payload with credentials", :type => :string, :default => '{}'
+  opt :inputs, "JSON payload to input to the workflow", :type => :string, :default => "{}"
+  opt :credentials, "JSON payload with credentials", :type => :string, :default => "{}"
+  opt :runner, "Type of runner for docker images", :type => :string, :default => "docker"
+  opt :runner_option, "Options to pass to the runner", :type => :strings
 end
 
 require "logger"
 Floe.logger = Logger.new(STDOUT)
 
 workflow = Floe::Workflow.load(opts[:workflow], opts[:inputs], opts[:credentials])
 
+runner_klass = case opts[:runner]
+               when "docker"
+                 Floe::Workflow::Runner::Docker
+               when "podman"
+                 Floe::Workflow::Runner::Podman
+               when "kubernetes"
+                 Floe::Workflow::Runner::Kubernetes
+               else
+                 raise
```

Comment:
The way I usually handle this is up above, after the Optimist.options, I do some sanity checks on the incoming parameters.  So in this case, I would do something like:

```ruby
opts = Optimist.options do
  # ...
end
Optimist.die(:runner, "must be one of #{Floe::Workflow::Runner::TYPES.join(", ")}) unless Floe::Workflow::Runner::TYPES.include?(opts[:runner])
```

where `Floe::Workflow::Runner::TYPES` looks like `%[docker podman kubernetes]`

You could probably do a similar sanity check that each of the runner_options looks like the `x=y` pattern.

Reply:
Optimist.die will give you a nicer error message and tell you to check --help and all that.